client/src/components/Header.js

```diff
@@ -60,23 +56,54 @@ export const ADMIN_LINKS = [
     href: "/admin/messages",
   },
 ];
-export default function Header({ links, isAdminPage }) {
+export default function Header({
+  links,
+  isAdminPage,
+  cartItems = [],
+  addToCart = () => {},
+}) {
   const { isOpen, onToggle } = useDisclosure();
   const { isAuthenticated, buildLogoutUrl, buildAuthorizeUrl } = useAuth0();
   const [authUrl, setAuthUrl] = useState("");
   useEffect(() => {
     (async () => {
       setAuthUrl(
         await (isAuthenticated
-          ? buildLogoutUrl() // this not to be waited
+          ? buildLogoutUrl({
+              returnTo: `${window.location.origin}/admin`,
+            }) // this not to be waited
           : buildAuthorizeUrl({
               // this returns a promise
               audience: process.env.REACT_APP_AUDIENCE,
             }))
       );
     })();
-  }, [isAuthenticated]); // when isAuthenticated changes logIn changes to logOut viseversa
+  }, [isAuthenticated, buildLogoutUrl, buildAuthorizeUrl]); // when isAuthenticated changes logIn changes to logOut viseversa
 
+  const itemCount = cartItems.reduce((acc, item) => {
```

Comment:
It may not be the best process of adding cart functionality in Header Component. I will keep it for now